### PR TITLE
countdown: retry delay

### PR DESCRIPTION
**Pull request by ben-bay**
Addresses issue #58.

Luckily, this is already built directly into celery: https://docs.celeryproject.org/en/latest/reference/celery.app.task.html 

Do we like `1` second as the default? Or would `0` be better?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Development dependencies install via pip: `pip install "merlinwf[dev]"`
 - `merlin status <yaml spec>` that returns queues, number of connected
   workers and number of unused tasks in each of those queues
+- The step/run field `countdown`, which adds a delay in seconds to retries.
```
Comment (lucpeterson) [inline CHANGELOG.md]: I think this name should be more descriptive like `retry_delay`

Comment (ben-bay) [reply CHANGELOG.md]: For some reason that's the name celery uses but we can change it.

Comment (lucpeterson) [reply CHANGELOG.md]: or retry_countdown would be good too

**Comment by lucpeterson** [comment]
we may need to pass the *args and **kwargs to the retry step

**Comment by lucpeterson** [comment]
and i'm not really sure if it's supposed to be raise self.retry or just self.retry

**Comment by ben-bay** [comment]
@lucpeterson implemented your suggestions from yesterday, but according to my tests this still isn't working properly. Debugging...

**Comment by lucpeterson** [comment]
what test are you doing?

**Comment by ben-bay** [comment]
> what test are you doing?

I'm just having a step `exit $(MERLIN_RESTART)` with `retry_delay` set to 100. When I monitor the restarts, they come a lot faster than they should.

**Pull request closed without merging**